ginga/rv/plugins/LineProfile.py

```diff
@@ -24,43 +24,46 @@ class LineProfile(GingaPlugin.LocalPlugin):
 
     Usage
     -----
-    1. Select an axis and pick a point using the cursor.
-    2. Left-click to mark position.
-    3. Use MultiDim to change step values of axes.
+    1. Select an axis.
+    2. Pick a point or draw a region using the cursor.
+    3. Use MultiDim to change step values of axes, if applicable.
     """
     def __init__(self, fv, fitsimage):
         super(LineProfile, self).__init__(fv, fitsimage)
 
         self.image = None
         self.layertag = 'lineprofile-canvas'
-        self.raster_file = False
-        self.pan2mark = False
-        self.wd = None
-        self.ht = None
         self.selected_axis = None
         self.hbox_axes = None
 
+        # For "marks" feature
+        self._new_mark = 'New'
+        self.mark_types = ['point', 'circle', 'rectangle', 'polygon']
+        self.mark_type = 'point'
+        self.mark_radius = 10  # For point only
+        self.mark_style = 'cross'  # For point only
+        self.mark_color = 'cyan'
+        self.marks = [self._new_mark]
+        self.mark_selected = self._new_mark
+        self.mark_index = 0
+        self.y_lbl = ''
+        self.x_lbl = ''
+
         self.dc = self.fv.get_draw_classes()
         canvas = self.dc.DrawingCanvas()
-        canvas.set_callback('cursor-down', self.btndown_cb)
-        canvas.set_callback('cursor-up', self.update)
+        canvas.enable_draw(True)
+        canvas.enable_edit(True)
+        canvas.set_drawtype(self.mark_type, color=self.mark_color,
+                            linestyle='dash')
+        canvas.set_callback('draw-event', self.draw_cb)
+        canvas.set_callback('edit-event', self.edit_cb)
+        canvas.add_draw_mode('move', down=self.buttondown_cb,
+                             move=self.motion_cb, up=self.buttonup_cb)
+        canvas.set_draw_mode('draw')
+        canvas.register_for_cursor_drawing(self.fitsimage)
         canvas.setSurface(self.fitsimage)
         self.canvas = canvas
 
-        # For "marks" feature
-        self.mark_radius = 10
-        self.mark_style = 'cross'
-        self.mark_color = 'purple'
-        self.select_color = 'cyan'
```

Comment:
I find having multiple colors to be quite confusing, especially when there are multiple plugins active. So, I removed one of the colors. Planning to make this color user-configurable.